context_test.go

```diff
@@ -180,6 +180,24 @@ func TestContextBuilderParentContextReplaceHeaders(t *testing.T) {
 		SetHeaders(map[string]string{"header key": "header value 2"}).
 		Build()
 	assert.Equal(t, map[string]string{"header key": "header value 2"}, ctx2.Headers())
+
+	goroutines.VerifyNoLeaks(t, nil)
+}
+
+func TestContextWrapWithHeaders(t *testing.T) {
+	headers1 := map[string]string{
+		"k1": "v1",
+	}
+	ctx, _ := NewContextBuilder(time.Second).
+		SetHeaders(headers1).
+		Build()
+	assert.Equal(t, headers1, ctx.Headers(), "Headers mismatch after Build")
+
+	headers2 := map[string]string{
+		"k1": "v1",
+	}
+	ctx2 := WrapWithHeaders(ctx, headers2)
+	assert.Equal(t, headers2, ctx2.Headers(), "Headers mismatch after WrapWithHeaders")
```

Comment:
Doesn't this test call `context.WithTimeout` in `Build()`, just like the other tests? We happen to know that, because of some implementation details, this *shouldn't* orphan a goroutine because (a) we're not setting a parent context and (b) we're now unwrapping the parent context.

Since we've introduced this ContextBuilder abstraction, I guess I'd expect the tests to assert that it's not leaking goroutines instead of assuming that some methods are implemented safely.